### PR TITLE
Add DenseNet

### PR DESCRIPTION
**Pull request by sachinprasadhs**
Add DenseNet Backbone and Classifier.

### CODE REVIEW

**Review by divyashreepathihalli — APPROVED** [review]
Thanks for the PR @sachinprasadhs!! Looks great!

The saved model test is failing with 
```
FAILED keras_nlp/src/models/densenet/densenet_image_classifier_test.py::DensekNetImageClassifierTest::test_saved_model - ValueError: Input 0 of layer "densek_net_image_classifier" is incompatible with the layer: expected shape=(None, 224, 224, 3), found shape=(2, 16, 16, 3)
```
To run the large tests(that is the saved model test) locally you can use `pytest <folder_path> --run_large`

**Comment by sachinprasadhs** [comment]
Thanks, update the test case.

File: keras_nlp/src/models/densenet/densenet_backbone.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        stackwise_num_repeats=[6, 12, 24, 16],
+        include_rescaling=False,
+    )
+    model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        *,
```
Comment (mattdangerw) [inline keras_nlp/src/models/densenet/densenet_backbone.py:58]: We don't have this leading start in other backbones.

File: keras_nlp/src/models/densenet/densenet_backbone.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        stackwise_num_repeats,
+        include_rescaling,
```
Comment (mattdangerw) [inline keras_nlp/src/models/densenet/densenet_backbone.py:60]: default this to true

File: keras_nlp/src/models/densenet/densenet_backbone.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+from keras_nlp.src.models.backbone import Backbone
+
+BN_AXIS = 3
+BN_EPSILON = 1.001e-5
+
+
+@keras_nlp_export("keras_nlp.models.DenseNetBackbone")
+class DenseNetBackbone(Backbone):
+    """Instantiates the DenseNet architecture.
+
```
Comment (mattdangerw) [inline keras_nlp/src/models/densenet/densenet_backbone.py:26]: add a short description

File: keras_nlp/src/models/densenet/densenet_image_classifier.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import keras
+
+from keras_nlp.src.api_export import keras_nlp_export
+from keras_nlp.src.models.densenet.densenet_backbone import DenseNetBackbone
+from keras_nlp.src.models.image_classifier import ImageClassifier
+
+
+@keras_nlp_export("keras_nlp.models.DensekNetImageClassifier")
+class DensekNetImageClassifier(ImageClassifier):
+    """DensekNet image classifier task model.
```
Comment (mattdangerw) [inline keras_nlp/src/models/densenet/densenet_image_classifier.py:23]: This looks copy and pasted directly from the github code viewer which can remove empty newlines. Restore them where applicable.

**Review by mattdangerw — COMMENTED** [review]
Thanks! A few minor comments.

File: keras_nlp/src/models/densenet/densenet_backbone.py (hunk 3 above)
Comment (sachinprasadhs) [reply keras_nlp/src/models/densenet/densenet_backbone.py:31]: Added

File: keras_nlp/src/models/densenet/densenet_backbone.py (hunk 2 above)
Comment (sachinprasadhs) [reply keras_nlp/src/models/densenet/densenet_backbone.py]: Done

File: keras_nlp/src/models/densenet/densenet_image_classifier.py (hunk 4 above)
Comment (sachinprasadhs) [reply keras_nlp/src/models/densenet/densenet_image_classifier.py]: Modified, Thanks

File: keras_nlp/src/models/densenet/densenet_image_classifier.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+class DenseNetImageClassifier(ImageClassifier):
+    """DenseNet image classifier task model.
+
+    Args:
+        backbone: A `keras_nlp.models.DenseNetBackbone` instance.
+        num_classes: int. The number of classes to predict.
+        activation: `None`, str or callable. The activation function to use on
+            the `Dense` layer. Set `activation=None` to return the output
+            logits. Defaults to `"softmax"`.
+    To fine-tune with `fit()`, pass a dataset containing tuples of `(x, y)`
```
Comment (mattdangerw) [inline keras_nlp/src/models/densenet/densenet_image_classifier.py:31]: Move this paragraph above args.

**Pull request merged**